recipe/build.sh

```diff
@@ -3,25 +3,27 @@ set -e
 set -x
 
 export CONFIGURE_ARGS="--with-eigen3=${PREFIX}/include/eigen3 ${CONFIGURE_ARGS}"
+export CONFIGURE_ARGS="-DCMAKE_C_COMPILER=${CC} -DCMAKE_CXX_COMPILER=${CXX}"
 
 if [[ -n "$mpi" && "$mpi" != "nompi" ]]; then
-  export CONFIGURE_ARGS="CC=mpicc CXX=mpic++ --with-mpi=${PREFIX} --with-zoltan="${PREFIX}" ${CONFIGURE_ARGS}"
+  export CONFIGURE_ARGS="-DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpic++ -DENABLE_MPI=ON -DENABLE_ZOLTAN=ON"
 fi
 
 if [[ -n "$tempest" && "$tempest" != "notempest" ]]; then
-  export CONFIGURE_ARGS="--with-tempestremap=${PREFIX} --with-netcdf=${PREFIX} ${CONFIGURE_ARGS}"
+  export CONFIGURE_ARGS="-DENABLE_TEMPESTREMAP=ON -DENABLE_NETCDF=ON ${CONFIGURE_ARGS}"
 fi
 
-autoreconf -fi
-./configure --prefix="${PREFIX}" \
-  ${CONFIGURE_ARGS} \
-  --with-hdf5="${PREFIX}" \
-  --with-metis="${PREFIX}" \
-  --enable-shared \
-  --enable-tools \
-  --enable-pymoab \
-  --disable-fortran \
-  || { cat config.log; exit 1; }
+mkdir bld
+cd bld
+cmake .. -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+         -DENABLE_HDF5=ON \
```

Comment:
It may be best to have the CMake workflow script side by side rather than replacing the workflow that has been verified for the entire build matrix. Especially if the main motivation is to get the build working on Windows first.

Reply:
I would be more inclined to switch over to cmake only if possible.

Reply:
Fair enough.  I'm going to continue testing with the CMake alone, however.  

(Our full software stack prefers to build with CMake on all platforms.)

Reply:
> I would be more inclined to switch over to use only cmake if possible.

That would also be my preference, but I don't think that CMake gets "first class" support with limited resources available on the team.  I've tried to keep it up-to-date for our limited use case, but it relies on a lot of (at-times-fragile) customization of package finding.

Reply:
Yeah thanks to Paul for keeping us focused on the CMake workflow even when we have been lagging support. I think it is in a decent enough state to replace autotools. I have my qualms about both workflows for configuration, but we will probably drop support for one of them soon (probably autotools). @gonuke I'll initiate a conversation in the mailing list this week.

The reason why I suggested having both workflows side by side is so that you can still get CMake working on Windows while keeping the rest of the platforms reliant on autotools for now since it's tested and verified already. And then we can make the transition along with MOAB when we deprecate autotools in the next major release. 

@iulian787 Thoughts ?

Reply:
@vijaysm and @gonuke, thank you both for clarifying.

Reply:
Building w/ LAPACK gets complicated on Windows, and therefore it's an obstacle for including tempest

Reply:
Would it be acceptable to include tempest and continue the move to cmake but just not build for windows (yet)? This would appear to still be a step on a good direction to me as we would still get cmake 🎉 . 

Then the complicated LAPACK and windows building can be addressed at another time in another PR